Mathlib/Tactic/CancelDenoms/Core.lean

```diff
@@ -160,9 +158,9 @@ def synthesizeUsingNormNum (type : Expr) : MetaM Expr := do
 canceled in `e'`, distributing `v` proportionally according to the tree `tr` computed
 by `findCancelFactor`.
 -/
-partial def mkProdPrf (α : Q(Type u)) (sα : Q(Field $α)) (v : ℕ) (t : Tree ℕ)
+partial def mkProdPrf {u : Level} (α : Q(Type u)) (sα : Q(Field $α)) (v : ℕ) (t : Tree ℕ)
     (e : Q($α)) : MetaM Expr := do
-  let amwo ← synthInstanceQ q(AddMonoidWithOne $α)
+  let amwo : Q(AddMonoidWithOne $α) := q(inferInstance)
```

Comment:
Is there a reason to prefer this over `synthInstanceQ`?

Reply:
Yes, this version does the instance search at compile time rather than each time the tactic is run.